### ISSUE TITLE
How does LifeWindow and CleanWindow work?

### ISSUE THREAD

**Issue by amudhangunasekaran** [allegro/bigcache#138]
I created a BigCache with only LifeWindow and the entries never got evicted. Came here to raise a bug, but realized that I should also set the CleanWindow from https://github.com/allegro/bigcache/issues/105.

But I still have a question. How does LifeWindow and CleanWindow work? If I set LifeWindow and CleanWindow as 10 seconds, will the entries get evicted 10 seconds after they were put in the cache or they will get evicted 10 seconds after they were last accessed? Is there any way I can set them so that the entries in the cache will be evicted 10 seconds after they were put in the cache no matter how frequently it was accessed?

Thanks and Regards,
Amudhan.

**Comment by tetafro** [comment]
https://github.com/allegro/bigcache/issues/123

**Comment by mxplusb** [comment]
@amudhangunasekaran `LifeWindow` is the lifecycle of the cache item, so how long before it gets evicted; `CleanWindow` is the frequency at which the evicted items are removed from memory. Does that answer your question?

**Comment by amudhangunasekaran** [comment]
@mxplusb thanks for your answer! It makes sense. I assume that the `LifeWindow` works on the insertion time and NOT on the last-accessed-time. This is the behavior I observed when I tested it.  And I assume that there is no way to configure that the cache entries should be evicted based on the last accessed time instead of the insertion time (this is not a problem for me and fits with my requirement).

Thanks and Best Regards!
